src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java

```diff
@@ -792,26 +802,44 @@ private void discoverLuckyOres()
                 }
 
                 final Item item = ForgeRegistries.ITEMS.getValue(new ResourceLocation(split[0]));
-                if (item == null || item == Items.AIR)
-                {
+                if (item == null || item == Items.AIR) {
                     Log.getLogger().warn("Invalid lucky block: " + ore);
                     continue;
                 }
 
+                final int defaultMineLevel = 0;
+                int buildingLevel = defaultMineLevel;
                 final ItemStack stack = new ItemStack(item, 1);
-                try
-                {
+                try {
+                    if (split.length == 3) {
+                        buildingLevel = Integer.parseInt(split[2]);
+                    }
+
                     final int rarity = Integer.parseInt(split[split.length - 1]);
-                    for (int i = 0; i < rarity; i++)
-                    {
-                        luckyOres.add(new ItemStorage(stack));
+
+                    luckyOres.putIfAbsent(buildingLevel, new ArrayList<>());
+
+                    for (int i = 0; i < rarity; i++) {
+                        List<ItemStorage> luckyOreOnLevel = luckyOres.get(buildingLevel);
+                        luckyOreOnLevel.add(new ItemStorage(stack));
                     }
+                } catch (final NumberFormatException ex) {
```

Comment:
This doesn't look according to the code style, please import the `documentation/formatter/intellij/Minecolonies.xml` into your editor (assuming you're working with IntelliJ)

Reply:
reformatted the code according to spec